tls/s2n_fingerprint.c

```diff
@@ -19,6 +19,164 @@
 #include "utils/s2n_mem.h"
 #include "utils/s2n_safety.h"
 
+static S2N_RESULT s2n_fingerprint_init(struct s2n_fingerprint *fingerprint,
+        s2n_fingerprint_type type)
+{
+    RESULT_ENSURE_REF(fingerprint);
+
+    switch (type) {
+        case S2N_FINGERPRINT_JA3:
+            fingerprint->method = &ja3_fingerprint;
+            break;
+        default:
+            RESULT_BAIL(S2N_ERR_INVALID_ARGUMENT);
+    }
+
+    const struct s2n_fingerprint_method *method = fingerprint->method;
+    RESULT_ENSURE_REF(method);
+    RESULT_GUARD_POSIX(s2n_hash_new(&fingerprint->hash));
+    s2n_hash_allow_md5_for_fips(&fingerprint->hash);
+    RESULT_GUARD_POSIX(s2n_hash_init(&fingerprint->hash, method->hash));
+    return S2N_RESULT_OK;
+}
+
+struct s2n_fingerprint *s2n_fingerprint_new(s2n_fingerprint_type type)
+{
+    DEFER_CLEANUP(struct s2n_blob mem = { 0 }, s2n_free);
+    PTR_GUARD_POSIX(s2n_alloc(&mem, sizeof(struct s2n_fingerprint)));
+    PTR_GUARD_POSIX(s2n_blob_zero(&mem));
+    struct s2n_fingerprint *fingerprint = (struct s2n_fingerprint *) (void *) mem.data;
+    PTR_ENSURE_REF(fingerprint);
+    PTR_GUARD_RESULT(s2n_fingerprint_init(fingerprint, type));
+    ZERO_TO_DISABLE_DEFER_CLEANUP(mem);
+    return fingerprint;
+}
+
+static S2N_CLEANUP_RESULT s2n_fingerprint_wipe(struct s2n_fingerprint *fingerprint)
```

Comment:
For this naming, `s2n_fingerprint_wipe` confused me for a split second because I assumed that it was a public method (since `s2n_connection_wipe` is a public method for similar functionality). Would it be more in line with conventions to have the public "this is now safe to reuse" method be `s2n_fingerprint_wipe`?

Maybe other developers will be quicker to notice the `static` and it won't be a problem, lol.